### DISCUSSION TITLE
Might Be Whistlin' Dixie but...

### DISCUSSION THREAD

**Discussion by wwderw** [david-vanderson/dvui#90] [Ideas]
I know Mach is trying this, I was just wondering.  They have forks of the typical system libraries (especially Linux where I do most of my stuff on) for X11, Wayland etc to have more portable builds.  Would that be something that we would be able to take advantage of to help out with more portable builds for this library?  On Linux, ideally to build against musl.

This would probably be more effective against the backends (I think Raylib is setup to build with the library (but I could be wrong), don't know about SDL).

I should warn everyone that I tend to know just enough to cause damage, so I don't know if I'm really out in left field with this or not.

Thoughts?  Stupid idea?

**Comment by david-vanderson** [comment]
Great idea - I would love to have this.  Unfortunately I don't really know how to make it happen.  My plan was to watch what the Mach people do and copy that.

DVUI is in a slightly different space, because we want to keep it easy to bolt onto an existing application/game.  So we need to take care to avoid progress here from complicating that usecase.  But it would be a huge win for people just starting out, especially if they just want to do a normal gui app.

Progress here will depend on someone willing to try something.

**Comment by nat3Github** [comment]
before i discovered dvui i was checking out mach. i think that was roughly 8 months ago and to be honest there were severe issues that prevented me from using it and worse it was not clear what features worked on which platform, havent checked lately if they made progress. afaik dvui is designed to work with different backends and mach wants to be modular so if its ready someone probably could write a backend using mach for dvui. the adaptability is a great feature of dvui imo! in the meantime there is nothing wrong using SDL and raylib they are heavily used in production and provide great backends! if mach turns out to support more platforms than i.e. SDL that would be a real motivation to look into it, right now - not so much.
